FlowCrypt/Controllers/Msg/MsgViewController.swift

```diff
@@ -58,8 +58,7 @@ final class MsgViewController: ASViewController<ASTableNode> {
         let buttons: [NavigationBarItemsView.Input]
         switch input?.path {
         case MailDestination.Gmail.trash.path: buttons = [infoInput, trashInput]
-        case MailDestination.Gmail.inbox.path: buttons = [infoInput, archiveInput, trashInput, mailInput]
-        default: buttons = [infoInput, trashInput, mailInput]
+        default: buttons = [infoInput, archiveInput, trashInput, mailInput]
```

Comment:
Because "archiveInput" will move move mail from inbox to "All Mail" (or more precisely, delete the inbox copy), it does not make sense to have the archive button on any mail outside of inbox.

Please look into it in more detail, to make sure the buttons in inbox differ from buttons in other folders, the way this switch case was originally written, and find out why it didn't work the way it was intended.

Reply:
You can look at what the Gmail app does, which is a good approximation of what we're trying to achieve when it comes to action buttons.

Reply:
You were right 
Google returns folders path as 

INBOX
[Gmail]/All Mail
[Gmail]/Drafts
[Gmail]/Important
[Gmail]/Sent Mail
[Gmail]/Spam
[Gmail]/Starred
[Gmail]/Trash

It leads to a bug with an archive button. Fixed in last commit